sr_port/hd.mpt

```diff
@@ -11,13 +14,17 @@
 %HD	;GT.M %HD utility - hexadecimal to decimal conversion program
 	;invoke at INT with %HD in hexadecimal to return %HD in decimal
 	;invoke at FUNC as an extrinsic function
+	;Note: Ignores any prefix of 0x or 0X
 	;if you make heavy use of this routine, consider $ZCALL
 	;
 	s %HD=$$FUNC(%HD)
 	q
 INT	r !,"Hexidecimal: ",%HD s %HD=$$FUNC(%HD)
 	q
 FUNC(h)
+	n prefix
+	set prefix=$zextract(h,1,2)
+	set:(("0X"=prefix)!("0x"=prefix)) h=($zextract(h,3,99))
 	q:$tr(h,"E","e")<0 ""
 	n c,d,dg
 	s d=0,h=$tr(h,"abcdef","ABCDEF")
```

Comment:
Instead of inserting lines 25 through 27, why not change line 30 to: ``s d=0,h=$tr(h,"abcdefx","ABCDEFX") s:"0X"=$e(h,1,2) $e(h,3,$l(h))``?

Reply:
Sure. Will do.